ui/pages/confirmations/components/confirm/splash/smart-account-update/smart-account-update.tsx

```diff
@@ -70,25 +79,32 @@ const ListItem = ({
 export function SmartAccountUpdate() {
   const [acknowledged, setAcknowledged] = useState(false);
   const t = useI18nContext();
+  const dispatch = useDispatch();
   const { currentConfirmation } = useConfirmContext<TransactionMeta>();
   const { handleRejectUpgrade } = useSmartAccountActions();
   const splashPageAcknowledgedForAccountList: string[] = useSelector(
     getUpgradeSplashPageAcknowledgedForAccounts,
   );
-
+  const smartAccountOptIn = useSelector(getUseSmartAccount);
   const { chainId, txParams, origin } = currentConfirmation ?? {};
   const { from } = txParams;
+  const account = useSelector((state: AccountsState) =>
+    getMemoizedInternalAccountByAddress(state as AccountsState, from),
+  );
+  const keyringType = account?.metadata?.keyring?.type;
```

Comment:
Do we need to reference the keyring type here?

We can't upgrade anything other than `hd` and `simple` so the users can only get to this via the manual flow.

Should we instead just update account details to disable the switch option for other keyring types?

Reply:
It is actually coming from requirement here:

> This opt in by default should only apply to fresh installs with newly generated SRPs. It should not apply to imported SRPs, imported private keys and connected hardware wallet accounts. For those, we should continue to ask the user to opt in at transaction/feature usage time.

https://github.com/MetaMask/MetaMask-planning/issues/4883

cc @bschorchit

Reply:
We can disregard that requirement and treat them all as accounts with exception of hardware wallets.